docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +27,180 @@ dependencies {
-The `@SpannerColumnInnerType` annotation is required for List properties due to Java's type erasure.
+- The `@Column` annotation optionally provides a different column name than that of the property.
+- Spanner supports `List` types as fields, the `@ColumnInnerType` annotation is required for `List` properties due to Java's type erasure.
+
+#### SpEL expressions for table names
+
+In some cases you might want the `@Table` table name to be determined dynamically. In this case, you can use https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#expressions[Spring Expression Language].
+
+For example:
+
+[source, java]
+----
+
+@Table(name = "#{'trades_'.concat(tableNameSuffix)}")
+public class Trade {
+// ...
+}
+----
+
+This table name will be resolved only if `tableNameSuffix` value/bean in the Spring `applicationContext` is defined, and then, for exapmle if it has the value "123", it will be `trades_123`.
+
+=== Template API
+
+`SpannerOperations` and its implementation, `SpannerTemplate` provides the XYZTemplate pattern familiar for Spring developers and as such provides:
+
+ - resource management
+ - one stop shop to Spanner operations with the Spring Data POJO mapping and conversion features
+ - exception conversion
+
+After your setup with Spring Boot Starter, you can configure a template simply by injecting a member of the `SpannerOperations` interface to your class:
+
+[source,java]
+----
+@SpringBootApplication
+public class SpannerTemplateExample {
+
+	@Autowired
+	SpannerOperations spannerTemplate;
+
+  public void doSomething() {
+			this.spannerOperations.delete(Trade.class, KeySet.all());
+      //...
+			Trade t = new Trade();
+      //...
+			this.spannerOperations.insert(t);
+      //...
+			List<Trade> tradesByAction = spannerOperations.findAll(Trade.class);
+      //...
+	}
+}
+----
+
+The Template API provides convenience methods for:
+
+- https://cloud.google.com/spanner/docs/reads[Reads], and by providing SpannerReadOptions and SpannerQueryOptions
+   ** Stale read
+   ** Read with secondary indices
+   ** Read with limits and flow control
+- https://cloud.google.com/spanner/docs/reads#execute_a_query[Queries]
+- DML operations (delete, insert, update, upsert)
+- partial reads: you can define a set of columns to be read into your entity
+- partial writes: if you have only a few properties that you want to persist from your entity, you can select those
+
+
+=== Supported Types
+
+Spring Data Spanner supports the following types for regular fields:
+
+* `com.google.cloud.ByteArray`
+* `com.google.cloud.Date`
+* `com.google.cloud.Timestamp`
+* `java.lang.Boolean`
+* `java.lang.Long`
+* `java.lang.String`
+* `double[]`
+* `long[]`
+* `boolean[]`
+
+Spring Data Spanner supports the following inner types for `List` fields:
+
+* `com.google.cloud.ByteArray`
+* `com.google.cloud.Date`
+* `com.google.cloud.Timestamp`
+* `java.lang.Boolean`
+* `java.lang.Long`
+* `java.lang.String`
+
+
+=== Repositories
```

Comment:
It's cool that you took the time to provide a glimpse of what Spring Data Repositories are, but I don't think that is on us.
You should have instead provided a short description of what they are, say we support them and then link to a Spring Data Repositories refdoc or so.